### ISSUE TITLE
False negative on incorrect override involving generic method

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#10197]
The following code sample should result in a type error, but it doesn't.

```python
class Base:
    def method1[T: Base](self, x: T) -> T: ...

class Derived(Base):
    def method1[T: Derived](self, x: T) -> T: ...
```

This was pointed out in [this thread](https://discuss.python.org/t/unsoundness-of-contravariant-self-type/86338).

Older versions of pyright (prior to 1.1.387) detected this error. It appears to be an unintentional regression related to [this change](https://github.com/microsoft/pyright/commit/84b77b03fc284c969811b6468c68f29ec95dfb62).

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.399.
